radicle-cli/src/git.rs

```diff
@@ -60,6 +60,35 @@ impl From<String> for Rev {
     }
 }
 
+#[derive(Clone)]
+pub struct Remote<'a> {
+    pub(crate) name: String,
+    pub(crate) url: radicle::git::Url,
+    inner: git2::Remote<'a>,
+}
+
+impl Remote<'_> {
+    pub fn refspecs(&self) -> git2::Refspecs {
+        self.inner.refspecs()
+    }
+}
+
+impl<'a> TryFrom<git2::Remote<'a>> for Remote<'a> {
+    type Error = anyhow::Error;
```

Comment:
I'd suggest adding a `RemoteError` type which has variants for `MissingUrl`, `MissingName`, and one for `ParseUrl`.

Reply:
we are talking about the `radicle::rad::RemoteError`? I see just this error kind

```
 253 #[derive(Error, Debug)]                                                                                                                                              | 228 pub fn rad_remote(repo: &Repository) -> anyhow::Result<(git2::Remote, Id)> {                                                                                          
 254 pub enum RemoteError { 2 implementations                                                                                                                             | 229     match radicle::rad::remote(repo) {                                                                                                                                
 255     #[error("git: {0}")]                                                                                                                                             | 230         Ok((remote, id)) => Ok((remote, id)),                                                                                                                         
 256     Git(#[from] git2::Error),                                                                                                                                        | 231         Err(radicle::rad::RemoteError::NotFound(_)) => Err(anyhow!(                                                                                                   
 257     #[error("invalid remote url: {0}")]                                                                                                                              | 232             "could not find radicle remote in git config; did you forget to run `rad init`?"                                                                          
 258     Url(#[from] transport::local::UrlError),                                                                                                                         | 233         )),                                                                                                                                                           
 259     #[error("invalid utf-8 string")]                                                                                                                                 | 234         Err(err) => Err(err).context("could not read git remote configuration"),                                                                                      
 260     InvalidUtf8,                                                                                                                                                     | 235     }                                                                                                                                                                 
 261     #[error("remote `{0}` not found")]                                                                                                                               | 236 }                                                                                                                                                                     
 262     NotFound(String),                                                                                                                                                | 237                                                                                                                                                                       
 263 } 
```

but yes I can use it

Reply:
I dunno what that `RemoteError` is. I was suggesting adding one in those module for this particular `Remote`

Reply:
Done! in the last commit!